torch/autograd/_functions/tensor.py

```diff
@@ -591,4 +591,24 @@ def backward(self, grad_output):
         return grad_input
 
 
+class Cumsum(Function):
+
+    def __init__(self, dim):
+        super(Cumsum, self).__init__()
+        self.dim = dim
+
+    def forward(self, input):
+        return torch.cumsum(input, dim=self.dim)
+
+    def backward(self, grad_output):
+        grad_input = torch.cumsum(-grad_output, dim=self.dim)
+
+        end_idx = grad_input.size(self.dim) - 1
+        index = (slice(None, None),) * self.dim + (slice(end_idx, end_idx + 1),)
```

Comment:
The whole point of `index` is that you now don't need that slice. However I just realized that it might be simpler to call `grad_input.narrow(self.dim, end_idx, 1)`.